pandas/core/indexes/period.py

```diff
@@ -306,7 +306,7 @@ def __contains__(self, key):
 
     @cache_readonly
     def _int64index(self):
-        return Int64Index(self.asi8, name=self.name, fastpath=True)
+        return Int64Index._simple_new(self.asi8, name=self.name)
```

Comment:
What is the performance implication of using _simple_new vs just Int64Index(...)?  If its sufficiently small, we should be using the public constructors

Reply:
@jbrockmendel this leads a bit further, but given that the public constructors (`__init__`) of Index classes are very complex (something we will not be able to change I think), I think that we need to see `_simple_new` as kind of a "internal public" method within the pandas code base.